src/client/opamInitDefaults.ml

```diff
@@ -19,6 +19,14 @@ let repository_url = {
 }
 
 let default_compiler =
+  OpamFormula.ors [
+    OpamFormula.Atom (OpamPackage.Name.of_string "ocaml-system",
+                      OpamFormula.Empty);
```

Comment:
What happens if the user has an old system compiler (e.g. they happen to be on a really old Ubuntu)? Does it quietly ignore it and move on to ocaml-base-compiler, or should we duplicate the `>= "4.05"` constraint here too?

Reply:
The former: this is in the tests here: https://github.com/ocaml/opam/pull/4607/files#diff-b3f6abdb5ab3eb6c5612e441c0709542022442b9cc6f47b5eae75bc5aa85799cR36-R54